app/helpers/title_and_tabset_helper.rb

```diff
@@ -239,46 +239,51 @@ def filter_for_type(types, type)
     types == [type] ? label : link_with_query(label, link, **help)
   end
 
-  # Sort links, for indexes
-  def add_sorter(query, links, link_all: false)
+  # Conditionally dds a group of sorting links, for indexes, if relevant
+  # These link back to the same index action, changing only the `by` param.
+  #
+  def add_sorter(query, sorts, link_all: false)
+    return unless sorts && (query&.num_results&.> 1)
+
     content_for(:sorter) do
-      if links && (query&.num_results&.> 1)
-        sorts = create_sorting_links(query, links, link_all)
+      links = create_sorting_links(query, sorts, link_all)
 
-        render(partial: "application/content/sorter", locals: { sorts: sorts })
-      else
-        ""
-      end
+      render(partial: "application/content/sorter", locals: { links: links })
     end
   end
 
-  # Create sorting links, "graying-out" the current order.
-  # Need query to know which is current order
-  # Links are arrays of [text, path]
-  def create_sorting_links(query, links, link_all)
-    results = []
-    this_by = (query.params[:by] || query.default_order).sub(/^reverse_/, "")
+  # Make HTML buttons after adding relevant info to the raw sorts
+  #
+  # The terminology we're using to build these may be confusing:
+  # `sorts` = the arrays of [by_param, :label.t] provided by index helpers.
+  # `sort_links` = the same arrays, turned into [:label.t, path, id, active].
+  # `links` = HTML links with all the fixin's, sent to the template
+  #
+  def create_sorting_links(query, sorts, link_all)
+    sort_links = assemble_sort_links(query, sorts, link_all)
 
-    links.each do |by, label|
-      results << link_or_grayed_text(link_all, this_by, label, query, by)
-    end
+    sort_links.map do |title, path, identifier, active|
+      classes = "btn btn-default"
+      classes += " active" if active
+      args = { class: class_names(classes, identifier) }
+      args = args.merge(disabled: true) if active
 
-    # Add a "reverse" button.
-    results << sort_link(:sort_by_reverse.t, query, reverse_by(query, this_by))
+      link_with_query(title, path, **args)
+    end
   end
 
-  def link_or_grayed_text(link_all, this_by, label, query, by)
-    if !link_all && (by.to_s == this_by)
-      [label.t, nil] # just text
-    else
-      sort_link(label.t, query, by)
```

Comment:
Note that a link with a `disabled` attribute is not clickable regardless of its `href`, so it's not necessary to send a blank path here.  I noticed the `link_or_grayed_text` method doesn't actually make a blank path — it makes a path to the index, with a `q` param but no `by` param.

I decided to handle it by setting a boolean `active` flag on the `sort_link` array. It's another attribute but it just seems valuable to preserve the correct destination path for the active link, for example if someone tries to "copy link" in the browser.